sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/statements.scala

```diff
@@ -179,6 +179,7 @@ case class AlterTableUnsetPropertiesStatement(
  */
 case class AlterTableSetLocationStatement(
     tableName: Seq[String],
+    partitionSpec: Option[TablePartitionSpec],
```

Comment:
I am piggybacking on the existing `AlterTableSetLocationStatement` (by looking at how `AlterTableSetLocationCommand` is doing). Please let me know if this is not what we want and I will create a separate statement for partition.